components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/listeners/KeyManagerUserOperationListener.java

```diff
@@ -61,32 +71,68 @@ public int getExecutionOrderId() {
     }
 
     /**
-     * Deleting user from the identity database prerequisites.
+     * Deleting user from the identity database prerequisites. Remove pending aprroval requests for the user and remove
+     * the gateway key chache.
      */
     @Override
-    public boolean doPreDeleteUser(String username, UserStoreManager userStoreManager)
-                                                                    throws UserStoreException {
+    public boolean doPreDeleteUser(String username, UserStoreManager userStoreManager) throws UserStoreException {
 
+        boolean isTenantFlowStarted = false;
+        ApiMgtDAO apiMgtDAO = new ApiMgtDAO();
+        try {
+            String tenantDomain = MultitenantUtils.getTenantDomain(APIUtil.replaceEmailDomainBack(username));
+
+            if (tenantDomain != null && !MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equals(tenantDomain)) {
+                isTenantFlowStarted = true;
```

Comment:
Fixed in https://github.com/praminda/carbon-apimgt/commit/06b8f56a8eb0fa81126c41fdce3f41e90a87b5c5